### ISSUE TITLE
Some tests fail with CMAKE_BUILD_TYPE=Release

### ISSUE THREAD

**Issue by reavertm** [vogel/injeqt#18]
When I build with -DCMAKE_BUILD_TYPE=Release, some tests fail.
Some not initialised variables perhaps?

```
maciek@liwardyna` /usr/local/portage/distfiles/injeqt-1.1.0/build $ ctest -VV --rerun-failed 
UpdateCTestConfiguration  from :/usr/local/portage/distfiles/injeqt-1.1.0/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/usr/local/portage/distfiles/injeqt-1.1.0/build/DartConfiguration.tcl
Test project /usr/local/portage/distfiles/injeqt-1.1.0/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start  1: /action-method-test

1: Test command: /usr/local/portage/distfiles/injeqt-1.1.0/build/test/action-method-test
1: Test timeout computed to be: 9.99988e+06
1: ********* Start testing of action_method_test *********
1: Config: Using QtTest library 5.7.1, Qt 5.7.1 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 4.9.4)
1: PASS   : action_method_test::initTestCase()
1: PASS   : action_method_test::should_create_empty()
1: PASS   : action_method_test::should_create_valid_from_tagged_action_method()
1: PASS   : action_method_test::should_create_valid_from_tagged_action_slot()
1: PASS   : action_method_test::should_invoke_have_results()
1: FAIL!  : action_method_test::should_throw_when_empty_method() Exception not thrown
1:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
1: FAIL!  : action_method_test::should_throw_when_arguments() Exception not thrown
1:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
1: FAIL!  : action_method_test::should_throw_when_arguments() Exception not thrown
1:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
1: FAIL!  : action_method_test::should_throw_when_invalid_tag() Exception not thrown
1:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
1: FAIL!  : action_method_test::should_throw_when_no_tag() Exception not thrown
1:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
1: FAIL!  : action_method_test::should_throw_when_signal() Exception not thrown
1:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
1: FAIL!  : action_method_test::should_throw_when_signal() Exception not thrown
1:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
1: FAIL!  : action_method_test::should_throw_when_constructor() Exception not thrown
1:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
1: PASS   : action_method_test::cleanupTestCase()
1: Totals: 6 passed, 8 failed, 0 skipped, 0 blacklisted, 0ms
1: ********* Finished testing of action_method_test *********
1/3 Test  #1: /action-method-test ..............***Failed    0.00 sec
test 21
    Start 21: /resolve-dependencies-test

21: Test command: /usr/local/portage/distfiles/injeqt-1.1.0/build/test/resolve-dependencies-test
21: Test timeout computed to be: 9.99988e+06
21: ********* Start testing of resolve_dependencies_test *********
21: Config: Using QtTest library 5.7.1, Qt 5.7.1 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 4.9.4)
21: PASS   : resolve_dependencies_test::initTestCase()
21: PASS   : resolve_dependencies_test::should_resolve_no_dependencies_when_no_objects_available()
21: FAIL!  : resolve_dependencies_test::should_resolve_all_dependencies() Compared values are not the same
21:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/resolve-dependencies-test.cpp(141)]
21: FAIL!  : resolve_dependencies_test::should_resolve_available_dependencies() Compared values are not the same
21:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/resolve-dependencies-test.cpp(165)]
21: FAIL!  : resolve_dependencies_test::should_resolve_available_dependencies_using_exact_matching() Compared values are not the same
21:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/resolve-dependencies-test.cpp(193)]
21: FAIL!  : resolve_dependencies_test::should_resolve_available_dependencies_using_exact_matching_not_using_subclass() Compared values are not the same
21:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/resolve-dependencies-test.cpp(223)]
21: PASS   : resolve_dependencies_test::should_resolve_available_dependencies_not_using_superclass()
21: PASS   : resolve_dependencies_test::cleanupTestCase()
21: Totals: 4 passed, 4 failed, 0 skipped, 0 blacklisted, 0ms
21: ********* Finished testing of resolve_dependencies_test *********
2/3 Test #21: /resolve-dependencies-test .......***Failed    0.00 sec
test 22
    Start 22: /setter-method-test

22: Test command: /usr/local/portage/distfiles/injeqt-1.1.0/build/test/setter-method-test
22: Test timeout computed to be: 9.99988e+06
22: ********* Start testing of setter_method_test *********
22: Config: Using QtTest library 5.7.1, Qt 5.7.1 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 4.9.4)
22: PASS   : setter_method_test::initTestCase()
22: PASS   : setter_method_test::should_create_empty()
22: PASS   : setter_method_test::should_create_valid_from_tagged_setter_method()
22: PASS   : setter_method_test::should_create_valid_from_tagged_setter_slot()
22: PASS   : setter_method_test::should_invoke_have_results()
22: FAIL!  : setter_method_test::should_throw_when_empty_method() Exception not thrown
22:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
22: FAIL!  : setter_method_test::should_throw_when_multiple_arguments() Exception not thrown
22:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
22: FAIL!  : setter_method_test::should_throw_when_invalid_tag() Exception not thrown
22:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
22: FAIL!  : setter_method_test::should_throw_when_no_tag() Exception not thrown
22:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
22: FAIL!  : setter_method_test::should_throw_when_signal() Exception not thrown
22:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
22: FAIL!  : setter_method_test::should_throw_when_constructor() Exception not thrown
22:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
22: FAIL!  : setter_method_test::should_throw_when_non_matching_parameters() Exception not thrown
22:    Loc: [/usr/local/portage/distfiles/injeqt-1.1.0/test/unit/expect.h(32)]
22: PASS   : setter_method_test::cleanupTestCase()
22: Totals: 6 passed, 7 failed, 0 skipped, 0 blacklisted, 0ms
22: ********* Finished testing of setter_method_test *********
3/3 Test #22: /setter-method-test ..............***Failed    0.00 sec

0% tests passed, 3 tests failed out of 3

Total Test time (real) =   0.01 sec

The following tests FAILED:
          1 - /action-method-test (Failed)
         21 - /resolve-dependencies-test (Failed)
         22 - /setter-method-test (Failed)
Errors while running CTest
```

**Comment by vogel** [comment]
Confirmed. Will be fixed for 1.2 (maybe tomorrow?)

**Comment by vogel** [comment]
Fixed in 1.2
